clientconn.go

```diff
@@ -1329,6 +1329,9 @@ func (ac *addrConn) createTransport(addr resolver.Address, copts transport.Conne
 	select {
 	case <-connectCtx.Done():
 		// We didn't get the preface in time.
+		// The error we pass to Close() is immaterial since there are no open
+		// streams at this point, so no trailers with error details will be sent
+		// out. We just need to pass a non-nil error.
```

Comment:
IMO you could just do:

```go
newTr.Close(transport.ErrConnClosing /* unused */)
// (or "unused as newTr could not have RPCs" if you feel like that's not explicit enough)
```

and accomplish the same thing as these 3 lines.

Reply:
I actually prefer the changes in this PR.
Explicit comments might help in the future given how forgetful humans are.